core/src/main/java/nl/nn/adapterframework/doc/DocWriter.java

```diff
@@ -0,0 +1,375 @@
+package nl.nn.adapterframework.doc;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import nl.nn.adapterframework.core.PipeRunException;
+import nl.nn.adapterframework.doc.objects.AClass;
+import nl.nn.adapterframework.doc.objects.AFolder;
+import nl.nn.adapterframework.doc.objects.AMethod;
+import nl.nn.adapterframework.doc.objects.BeanProperty;
+import nl.nn.adapterframework.doc.objects.IbisBean;
+import nl.nn.adapterframework.doc.objects.MethodExtra;
+import nl.nn.adapterframework.doc.objects.DocInfo;
+import nl.nn.adapterframework.util.XmlBuilder;
+
+public class DocWriter {
+	public static String getSchema(DocInfo docInfo) throws PipeRunException {
+		XmlBuilder schema;
+		XmlBuilder element;
+		XmlBuilder complexType;
+		XmlBuilder choice;
+
+		schema = new XmlBuilder("schema", "xs", "http://www.w3.org/2001/XMLSchema");
+		schema.addAttribute("xmlns:xs", "http://www.w3.org/2001/XMLSchema");
+		schema.addAttribute("elementFormDefault", "qualified");
+
+		element = new XmlBuilder("element", "xs", "http://www.w3.org/2001/XMLSchema");
+		element.addAttribute("name", "Configuration");
+		element.addAttribute("type", "ConfigurationType");
+		schema.addSubElement(element);
```

Comment:
van deze vier regels kan je een method addElement() maken, waardoor het uiteindelijk veel duidelijker wordt wat er in getSchema() gebeurt.